llvm/lib/Target/RISCV/Disassembler/RISCVDisassembler.cpp

```diff
@@ -591,6 +599,30 @@ static DecodeStatus decodeRVCInstrRdRs1ImmZero(MCInst &Inst, uint32_t Insn,
   return S;
 }
 
+static DecodeStatus decodeRVCInstrShiftLeftHint(MCInst &Inst, uint32_t Insn,
+                                                uint64_t Address,
+                                                const MCDisassembler *Decoder) {
+  DecodeStatus S = MCDisassembler::Success;
+  uint32_t Rd = fieldFromInstruction(Insn, 7, 5);
+  if (!Check(S, DecodeGPRRegisterClass(Inst, Rd, Address, Decoder)))
+    return MCDisassembler::Fail;
+  Inst.addOperand(Inst.getOperand(0));
+  Inst.addOperand(MCOperand::createImm(0));
+  return S;
+}
+
+static DecodeStatus
+decodeRVCInstrShiftRightHint(MCInst &Inst, uint32_t Insn, uint64_t Address,
+                             const MCDisassembler *Decoder) {
+  DecodeStatus S = MCDisassembler::Success;
+  uint32_t Rd = fieldFromInstruction(Insn, 7, 3);
+  if (!Check(S, DecodeGPRCRegisterClass(Inst, Rd, Address, Decoder)))
+    return MCDisassembler::Fail;
+  Inst.addOperand(Inst.getOperand(0));
+  Inst.addOperand(MCOperand::createImm(0));
+  return S;
+}
```

Comment:
I hope at some point we can get rid of a lot of these custom full-instruction decoders, maybe by making per-operand decoders for specific operands? This doesn't need to be done in this change.